### ISSUE TITLE
Is it possible to save the certificate with code?

### ISSUE THREAD

**Issue by yuvalpikel** [justcoding121/Titanium-Web-Proxy#737]
Hi

Is it possible to load the certificate using code ?
for the first time the app is running it asked if i am sure that i want to install the certificate

Regards

**Comment by honfika** [comment]
Yes, of course it is possible.. jsut set in in the certifiatemanager.

**Comment by yuvalpikel** [comment]
Thanks.. In which methods / properties should i use?

בתאריך יום ה׳, 23 בינו' 2020, 8:18, מאת honfika ‏<notifications@github.com>:

**Comment by yuvalpikel** [comment]
Hi ..
I used : proxyServer.CertificateManager.LoadRootCertificate(@"C:\temp1\rootCert1.pfx", "");
But it still shows the message .. any ideas?

**Comment by honfika** [comment]
is that certificate already trusted on your machine?

generating a new certificate is not showing any message... it is needed for the trust... so if you load a certificate which is untused, it still needs to show that message.

**Comment by yuvalpikel** [comment]
Understood..
For which site the certificate should be created by CA?

בתאריך יום ה׳, 23 בינו' 2020, 11:26, מאת honfika ‏<notifications@github.com

**Comment by honfika** [comment]
I don't understand your question. Certificate is needed if you want to dertypt the https traffic

**Comment by yuvalpikel** [comment]
I understood that .. but what are the parameters that i should use in the
certificate?
If i need a trusted cert is should be issued by CA right?

**Comment by honfika** [comment]
Sorry, I don't know this... you should ask your infrastructure administrator.

**Comment by yuvalpikel** [comment]
Thanks...
But the code is a i'v wrote?

Yuval

**Comment by honfika** [comment]
LoadRootCertificate? Yes

**Comment by justcoding121** [comment]
Closing this as answered. Use CertificateManager.LoadRootCertificate(pfxPath, password) before starting the proxy. Loading a PFX and trusting it are separate operations: the client machine must already trust that root, or installation will still require the appropriate user/admin approval. In managed environments, deploy the root through the organization's certificate policy rather than suppressing the trust prompt in application code.
